scripts/train/train.py

```diff
@@ -164,170 +165,315 @@ def build_dataloader(cfg: DictConfig, tokenizer: PreTrainedTokenizerBase,
+    save_filename: str = pop_config(
+        cfg,
+        'save_filename',
+        must_exist=False,
+        default_value='ep{epoch}-ba{batch}-rank{rank}.pt')
+    save_interval: Union[str, int] = pop_config(cfg,
+                                                'save_interval',
+                                                must_exist=False,
+                                                default_value='1000ba')
+    save_num_checkpoints_to_keep: int = pop_config(
+        cfg, 'save_num_checkpoints_to_keep', must_exist=False, default_value=-1)
+    progress_bar = pop_config(cfg,
+                              'progress_bar',
+                              must_exist=False,
+                              default_value=False)
+    log_to_console: bool = pop_config(cfg,
+                                      'log_to_console',
+                                      must_exist=False,
+                                      default_value=True)
+    python_log_level: str = pop_config(cfg,
+                                       'python_log_level',
+                                       must_exist=False,
+                                       default_value='debug')
+    console_log_interval: Union[int, str] = pop_config(cfg,
+                                                       'console_log_interval',
+                                                       must_exist=False,
+                                                       default_value='1ba')
+    device_train_microbatch_size: Union[str, int] = pop_config(
+        cfg,
+        'device_train_microbatch_size',
+        must_exist=False,
+        default_value='auto')
+    eval_subset_num_batches: int = pop_config(cfg,
+                                              'eval_subset_num_batches',
+                                              must_exist=False,
+                                              default_value=-1)
+    eval_first: bool = pop_config(cfg,
+                                  'eval_first',
+                                  must_exist=False,
+                                  default_value=False)
+    load_path: str = pop_config(cfg,
+                                'load_path',
+                                must_exist=False,
+                                default_value=None)
+    load_weights_only: bool = pop_config(cfg,
+                                         'load_weights_only',
+                                         must_exist=False,
+                                         default_value=False)
+    load_ignore_keys: Optional[List[str]] = pop_config(cfg,
+                                                       'load_ignore_keys',
+                                                       must_exist=False,
+                                                       default_value=None)
+    # Enable autoresume from model checkpoints if possible
+    autoresume_default = not save_overwrite and not save_weights_only
```

Comment:
I don't think this is the same default condition as before

Reply:
ah good catch